includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -387,7 +387,8 @@ private function process_webhook_payment_intent_succeeded( $event_body ) {
 
 		$order_changed = false;
 		foreach ( $meta_data_to_update as $key => $value ) {
-			if ( $value && ! $order->get_meta( $key ) ) {
+			// Override existing meta data with incoming values, if present.
+			if ( $value ) {
 				$order_changed = true;
```

Comment:
@deepakpathania Is `$order_changed` required since we explicitly update the incoming metadata? Also, we're not comparing whether the value has changed.

Reply:
It is unlikely, but if we get a webhook event for this where the `value` isn't present for these keys, then we will skip the metadata update in the loop and in that case the order save call can be avoided.

Reply:
Let's remove it @deepakpathania. Like you said it's highly unlikely that all the keys would be empty and we do have a value check to prevent clearing the existing value.